octis/models/base_etm.py

```diff
@@ -45,24 +46,35 @@ def preprocess(vocab2id, train_corpus, test_corpus=None, validation_corpus=None)
         raise NotImplementedError("Subclasses should implement this!")
 
     def load_embeddings(self):
-        if not self.hyperparameters['train_embeddings']:
-            vectors = {}
-            embs = pkl.load(open(self.hyperparameters['embeddings_path'], 'rb'))
-            for l in embs:
-                line = l.split()
-                word = line[0]
-                if word in self.vocab.values():
-                    vect = np.array(line[1:]).astype(np.float)
-                    vectors[word] = vect
-            embeddings = np.zeros((len(self.vocab.keys()), self.hyperparameters['embedding_size']))
-            words_found = 0
-            for i, word in enumerate(self.vocab.values()):
-                try:
-                    embeddings[i] = vectors[word]
-                    words_found += 1
-                except KeyError:
-                    embeddings[i] = np.random.normal(scale=0.6, size=(self.hyperparameters['embedding_size'],))
-            self.embeddings = torch.from_numpy(embeddings).to(self.device)
+        if self.hyperparameters['train_embeddings']:
+            return
+        
+        vectors = self._load_word_vectors(self.hyperparameters['embeddings_path'], 
+                                        self.hyperparameters['embeddings_type'],
+                                        self.hyperparameters['binary_embeddings'])
+        embeddings = np.zeros((len(self.vocab.keys()), self.hyperparameters['embedding_size']))
+        for i, word in enumerate(self.vocab.values()):
+            try:
+                embeddings[i] = vectors[word]
+            except KeyError:
+                embeddings[i] = np.random.normal(scale=0.6, size=(self.hyperparameters['embedding_size'],))
+        self.embeddings = torch.from_numpy(embeddings).to(self.device)
+
+    def _load_word_vectors(self, embeddings_path, embeddings_type, binary_embeddings=True):
```

Comment:
Can you add the comments to explain the parameters of the method? Including the type and default values

Reply:
Done